src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/vehicle_info_plugin.cc

```diff
@@ -158,7 +173,33 @@ void VehicleInfoPlugin::DeleteSubscriptions(
     }
   }
 }
+
+void VehicleInfoPlugin::SelectVehicleDataForUnsubscribe(
+    const app_mngr::ApplicationSharedPtr& app,
+    std::set<mobile_apis::VehicleDataType::eType>& vehicle_data_for_unscribe) {
+  VehicleInfoAppExtension& vehicle_extension =
+      VehicleInfoAppExtension::ExtractVIExtension(*app);
+  vehicle_data_for_unscribe = vehicle_extension.Subscriptions();
+  application_manager::ApplicationSet applications =
+      application_manager_->applications().GetData();
+  if (!applications.empty()) {
+    auto unsubscriber = [&applications, &vehicle_data_for_unscribe](
+        const VehicleDataType vehicle_data) {
+      for (auto app_it = applications.begin(); app_it != applications.end();
```

Comment:
@IGapchuk you could use range-based loop here

Reply:
Done in 73c67bf4f764f384b64cc45f9a7a3087eff31f1f.